pkg/enemy/attack.go

```diff
@@ -155,29 +165,65 @@ func (e *Enemy) attack(atk *combat.AttackEvent, evt glog.Event) (float64, bool)
 						Ele:    e.prof.ParticleElement,
 					})
 				},
-				100, //TODO: should be subject to global delay maybe??
+				100, // TODO: should be subject to global delay maybe??
 			)
 		}
 	}
 
+	if e.prof.ParticleDrops != nil {
+		e.tryHPDropParticle()
+	}
+
```

Comment:
NIT: is there a way we can merge the `ParticleDropThreshold` logic with `tryHPDropParticle`? Looks like your current logic makes these two settings mutually exclusive but wondering if there is a better way which could simplify logic here

Reply:
With the struct processing from above, I think fixing this is relatively easy. Working on it.